Bugzilla/JobQueue/Runner.pm

```diff
@@ -58,20 +67,23 @@ sub gd_getopt {
         $self->{gd_progname} = $self->{gd_args}{progname};
     }
     else {
-        $self->{gd_progname} = basename($0);
+        $self->{gd_progname} = basename($PROGRAM_NAME);
     }
 
-    # There are places that Daemon Generic's new() uses $0 instead of
+    # There are places that Daemon Generic's new() uses $PROGRAM_NAME instead of
     # gd_progname, which it really shouldn't, but this hack fixes it.
-    $self->{_original_zero} = $0;
-    $0 = $self->{gd_progname};
+    $self->{_original_program_name} = $PROGRAM_NAME;
+
+    ## no critic (Variables::RequireLocalizedPunctuationVars)
```

Comment:
this disables this perlcritic policy until the end of the file; add a `## use critic` to re-enable (or move `## no critic` onto the same line as the assignment to `$PROGRAM_NAME`)